protos/common.proto

```diff
@@ -40,8 +48,12 @@ message Unit {
   string callsign = 3;
   Coalition coalition = 4;
   string type = 5;
-  Position position = 6;
```

Comment:
I was wondering whether we should only provide non-static values via dedicated RPC calls. Which would be `GetUnitPosition` in this case.

Reply:
Is the thinking for this due to performance reasons? I am honestly not sure how much extra time it takes to return a full unit object vs just the position but I am not sure it would be worth separate calls..

Reply:
I personally don't really see a need for a separate "position" call but that is based on my thinking that the amount of time it takes to create a position response vs a full Unit response object is negligable.

However I don't mind having a separate request _in addition_ to including the position in a full Unit response object. But I am not in favour of _requiring_ a separate call.

Reply:
I personally don't really see a need for a separate "position" call but that is based on my thinking that the amount of time it takes to create a position response vs a full Unit response object is negligable.

However I don't mind having a separate request in _addition_ to including the position in a full Unit response object since it lets the client decide which they needed. But I am not in favour of _requiring_ a separate call.

Do we need a generic `GetUnit` call as well? Right now we can get the units in a group and get the units position but cannot get the full info on an individual unit.

Reply:
While I would probably tend to only expose the position via an additional RPC call in a Rust client, having the position in both the full Unit response object and the separate call is actually still fine for me. Could probably be useful in context of an event to know the position at the time of the event.

```rust
struct Unit {
  position: Position,
}

impl Unit {
  fn get_position(&self) -> Position {
    // what to return? The potentially outdated `self.position`, or the result of an RPC call to `GetUnitPosition`?
  }
}
```

The reason I'd prefer the separate call in a client is that I wouldn't know what to return when the client user asks for the position. Should I get the most recent position or return a potentially outdated one?

> Do we need a generic `GetUnit` call as well? Right now we can get the units in a group and get the units position but cannot get the full info on an individual unit. (Although that might not be needed for this so I can add that later)

I'd say no, mainly because I am not aware of a DCS Lua method to get a specific unit by name (we can only get units by going through a group AFAIK).

The question is also, what is the full response? I'd tend towards requiring a separate RPC call for all information that might change like position, health, ammo etc. instead of one that returns all of them.

Reply:
I agree that we don't want a call that returns *everything* for a unit but position is the exception. Unit position It is so integral to every use-case I can think of for unit streaming use-cases that would use this that it should be included.

If a client specifically calls the `get_position` function then I do think that it should return the results of a `GetUnitPosition` call and not `self.position`. If a client is specifically asking for the position of a unit then chances are it is not actually using the streaming API at all otherwise it would already have that info.

Although in all honesty, I cannot really think of a situation where manually getting groups, units and unit positions is more efficient than just receiving the data via the streaming API so we might be in a case of discussing something that is YAGNI (You ain't gonna need it) anyway.